csrc/preseg_passes/insert_reshardings.cpp

```diff
@@ -55,8 +52,10 @@ void insertReshardingsBefore(Fusion* fusion) {
     auto output = expr->output(0)->as<TensorView>();
 
     std::unordered_set<TensorView*> inputs;
+    IdModel id_model({expr}, {}, false, false);
+    id_model.buildPermissiveGraph();
```

Comment:
I think either `Exact` or `Broadcast` should be used. The mapping sets are ordered as `Exact <= Broadcast <= Permissive`, where `Permissive` is the largest set containing all of the mappings of `Broadcast` (and thus `Exact`) with some more additional mappings based on inlining, but there's no inlining at this stage, so `Broadcast` or `Exact` should be enough.

The difference of `Exact` and `Broadcast` is when there's an expr like `t2[i2] = t0[i0] + t1[b1]`, i.e., just an addition of two 1D tensors, one of which is a broadcast tensor. In the `Exact` graph, only `i0` and `i2` are mapped, whereas in the `Broadcast` graph, all of `i0`, `b1` and `i2` are mapped.

Whether broadcast IDs should be mapped or not would depend on each use case.

Reply:
Or maybe `AlmostExact` instead of `Exact`. In `AlmostExact`, when an ID is merged with a broadcast ID of size 1, the output ID of the merge is mapped with the input ID.

Reply:
Got it! Will try AlmostExact...

Reply:
ALMOSTEXACT fails ReshardingTest.Add_Broadcast, and BROADCAST fails NonreshardingSqueeze. Is that a good reason to keep using PERMISSIVE?

Reply:
Can you give more details? How did they fail?

Reply:
Sure. I was referring to the new unit tests added in this PR. 

> ALMOSTEXACT fails Add_Broadcast

I believe this is because ALMOSTEXACT doesn't map broadcast to non-broadcast. 

> BROADCAST fails NonreshardingSqueeze

I believe this is because BROADCAST doesn't map i0*b1 to i0.

Reply:
Oh, I see. I think AlmostExact makes most sense. That's we use for the intra-device sync analysis. It doesn't work for cases like `Add_Broadcast`. We could create another graph that takes a union of AlmostExact and Broadcast, which should work for both cases. For the intra-device sync analysis, we check if the corresponding producer ID is a broadcast: https://github.com/NVIDIA/Fuser/blob/main/csrc/device_lower/analysis/sync_information.cpp#L721-L766 I think we could do something similar here too.

I understand that Permissive works too, but that graph is mainly for building the loop graph and we may decide not to expose it in the future. So if we could use some other graph, that would be better.